s4e-backend/src/main/java/pl/cyfronet/s4e/data/repository/query/QueryEnding.java

```diff
@@ -0,0 +1,43 @@
+package pl.cyfronet.s4e.data.repository.query;
+
+import java.util.List;
+import java.util.Map;
+
+public class QueryEnding extends QueryDecorator {
+    public QueryEnding(QueryBuilder queryBuilder) {
+        super(queryBuilder);
+    }
+
+    @Override
+    public void doPrepareQueryAndParameters(Map<String, Object> params,
+                                            List<Object> parameters,
+                                            StringBuilder resultQuery) {
+        if (params.containsKey("orderBy") && params.containsKey("sortBy")) {
+            resultQuery.append(" ORDER BY " + getOrderByField(params) + " " + getOrderBy(params));
+        }
+        resultQuery.append(" LIMIT ? ");
+        parameters.add(getLimit(params));
+        resultQuery.append(" OFFSET ? ;");
+        parameters.add(Integer.parseInt(params.getOrDefault("offset", 0).toString()));
+    }
+
+    private String getOrderByField(Map<String, Object> params) {
+        switch (params.get("sortBy").toString()) {
+            case "ingestion_time":
+                return "ingestion_time";
+            case "sensing_time":
+                return "sensing_time";
+            default:
+                return "id";
+        }
+    }
+
+    private String getOrderBy(Map<String, Object> params) {
+        return params.get("orderBy").equals("ASC") ? "ASC" : "DESC";
+    }
+
+    private int getLimit(Map<String, Object> params) {
+        int result = Integer.parseInt(params.getOrDefault("limit", 100).toString());
+        return result < 1000 ? result : 1000;
```

Comment:
Wouldn't it be better to use the values from `SpringDataWebProperties`? We're using them for other Pageable, might be sensible to use same values here, or at least not hardcode them.